docs/xbcloud-binary-fifo-datasink.md

```diff
@@ -0,0 +1,64 @@
+
+This is how it works:
+
+XtraBackup spawns multiple copy threads and each copy thread reads a data chunk from a specific file. Then each copy thread writes the data chunks to a pipe (STDOUT). An xbcloud read thread reads each chunk of STDIN data. Then xbcloud uploads the chunks to an object storage using an async request, and adds s callback to an event handler list. The xbcloud event handler executes the callback depending on the response from the object storage (success or failure).
+
+![image](_static/backup-streamed-to-object-storage.png)
+
+XtraBackup streaming has a limitation of 1.8 Gbps. Xtrabackup uses multiple copy threads to read data chunks from files in parallel, but the threds are serialized when Xtrabackup writes them to STDOUT. Uploading chunks on a Wide Area Network (WAN) experiences no lag, but streaming backups using a Local Area Network (LAN) can experience a lag because those configurations are designed for at least 10Gbps.
+
+Starting with [Percona XtraBackup 8.0.33-28](release-notes/8.0/8.0.33-28.0.md), XtraBackup can stream data to `xbcloud` using `FIFO` (named pipes).
+
+This is how it works:
+
+XtraBackup spawns multiple copy threads and each copy thread reads a data chunk from a specific file. Then multiple FIFO files are created to store the data from XtraBackup. Each xtrabackup copy thread writes the data chunks to a specific FIFO file. Xbcloud creates multiple pairs of read threads and event handlers to read from FIFO. Then xbcloud uploads data to object storage using async request. The xbcloud event handler executes the callback depending on the response from the object storage (success or failure). 
+
+![image](_static/fifo-datasink.png)
+
+## FIFO data sink options
+
+[Percona XtraBackup 8.0.33-28](release-notes/8.0/8.0.33-28.0.md) implements the following options:
+
+* `--fifo-streams=#` - specifies the number of FIFO files to use for parallel data stream. To disable FIFO data sink and send stream to STDOUT, set `--fifo-streams=1`. The default value is `1` (STDOUT) to ensure the backward compatibility.
+* `--fifo-dir=name` - specifies a directory to write Named Pipe.
+* `--fifo-timeout=#` - specifies the number of seconds to wait for the other end to open the stream for reading. The default value is `60` seconds.
+
+## How to run FIFO data sink
+
+To run FIFO data sink, you need to either invoke the following commands in separate terminal sessions: 
+
+```{.bash data-prompt="$"}
+$ xtrabackup --backup --stream --fifo-streams=2 --fifo-dir=/tmp/stream
+```
+
+```{.bash data-prompt="$"}
+$ xbcloud put --fifo-streams=2 --fifo-dir=/tmp/stream
+```
+
+or run xtrabackup in the background.
+
+### Performance improvement 
+
+Here is the example of using FIFO data sink compared to STDOUT method:
+
+We create 1TB of data in multiple tables. The link speed between the source server and destination server using MinIO is ~ 9.2 Gbps.
```

Comment:
The database has 1TB of data in multiple tables.

Reply:
fixed